tests/integration/test_charm.py

```diff
@@ -280,3 +281,58 @@ async def test_seldon_deployment(ops_test: OpsTest):
     assert response["data"]["names"] == ["proba"]
     assert response["data"]["tensor"]["shape"] == [2, 1]
     assert response["meta"] == {}
+
+
+def is_empty(iterator_list):
+    """Return True if supplied iterator is empty, False otherwise."""
+    try:
+        _ = next(iterator_list)
+    except StopIteration:
+        # iterator is empty
+        return True
+    return False
+
+
+@pytest.mark.abort_on_fail
+async def test_remove_with_resources_present(ops_test: OpsTest):
+    """Test remove with all resources deployed.
+
+    Verify that all deployed resources that need to be removed are removed.
+
+    This test should be before test_upgrade(), because it removes deployed charm.
+    """
+    # remove deployed charm and verify that it is removed
+    await ops_test.model.remove_application(app_name=APP_NAME, block_until_done=True)
+    assert APP_NAME not in ops_test.model.applications
+
+    # verify that all resources that were deployed are removed
+    lightkube_client = Client()
+
+    # verify all CRDs in namespace are removed
+    crd_list = lightkube_client.list(
+        CustomResourceDefinition,
+        labels=[("app.juju.is/created-by", "seldon-controller-manager")],
+        namespace=ops_test.model.name,
+    )
+    assert is_empty(crd_list)
```

Comment:
I would recommend to rewrite this pattern. My rationale is the following:
1. we can make it more readable
2. we exclude bad side effects where user can reuse iterator, but you already poped one item (see my example)
```
c = Counter(3, 5)
next(c)
# >>> 3
list(c)
# >>> [4]
```
vs

```
c = Counter(3, 5)
list(c)
# >>> [3, 4]
```

```suggestion
    assert not list(crd_list)
```

Reply:
Will do.